File: Angular/src/app/models/packages/saving.package.ts
import { CommunicationPackage } from "./communication.package";
import { SettingsPackage } from "./settings.package";
import { TagsPackage } from "./tags.package";

export interface SavingPackage extends CommunicationPackage {

    settingsPackage: SettingsPackage;
    tagsPackage: TagsPackage;
}

File: Spring/src/main/kotlin/com/example/spring/models/packages/SavingPackage.kt
package com.example.spring.models.packages

import com.google.gson.annotations.SerializedName

data class SavingPackage(

    @SerializedName("Settings")
    val settingsPackage: SettingsPackage = SettingsPackage(),

    @SerializedName("Tags")
    val tagsPackage: TagsPackage = TagsPackage()
): CommunicationPackage()
